### ISSUE TITLE
Package collides with custom form authenticator on Symfony 5

### ISSUE THREAD

**Issue by KepplerPl** [lexik/LexikJWTAuthenticationBundle#717]
I'm using Symfony 5 with this package. I've installed it using the instructions provided in the readme.

The packge itself works fine.

I have a custom form authenticator `LoginFormAuthenticator.php`

```
<?php

namespace App\Security;

use App\Entity\User;
use Doctrine\ORM\EntityManagerInterface;
use Symfony\Component\HttpFoundation\RedirectResponse;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
use Symfony\Component\Routing\RouterInterface;
use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
use Symfony\Component\Security\Core\Exception\CustomUserMessageAuthenticationException;
use Symfony\Component\Security\Core\Exception\InvalidCsrfTokenException;
use Symfony\Component\Security\Core\Security;
use Symfony\Component\Security\Core\User\UserInterface;
use Symfony\Component\Security\Core\User\UserProviderInterface;
use Symfony\Component\Security\Csrf\CsrfToken;
use Symfony\Component\Security\Csrf\CsrfTokenManagerInterface;
use Symfony\Component\Security\Guard\Authenticator\AbstractFormLoginAuthenticator;
use Symfony\Component\Security\Guard\PasswordAuthenticatedInterface;
use Symfony\Component\Security\Http\Util\TargetPathTrait;

class LoginFormAuthenticator extends AbstractFormLoginAuthenticator implements PasswordAuthenticatedInterface
{
    use TargetPathTrait;

    private $entityManager;
    private $urlGenerator;
    private $csrfTokenManager;
    private $passwordEncoder;

    public function __construct(EntityManagerInterface $entityManager, UrlGeneratorInterface $urlGenerator, CsrfTokenManagerInterface $csrfTokenManager, UserPasswordEncoderInterface $passwordEncoder)
    {
        $this->entityManager = $entityManager;
        $this->urlGenerator = $urlGenerator;
        $this->csrfTokenManager = $csrfTokenManager;
        $this->passwordEncoder = $passwordEncoder;
    }

    public function supports(Request $request)
    {
        return 'login_user' === $request->attributes->get('_route') && $request->isMethod('POST');
    }

    public function getCredentials(Request $request)
    {
        $credentials = [
            'email' => $request->request->get('email'),
            'password' => $request->request->get('password'),
            'csrf_token' => $request->request->get('_csrf_token'),
        ];
        $request->getSession()->set(
            Security::LAST_USERNAME,
            $credentials['email']
        );

        return $credentials;
    }

    public function getUser($credentials, UserProviderInterface $userProvider)
    {
        $token = new CsrfToken('authenticate', $credentials['csrf_token']);
        if (!$this->csrfTokenManager->isTokenValid($token)) {
            throw new InvalidCsrfTokenException();
        }

        $user = $this->entityManager->getRepository(User::class)->findOneBy(['email' => $credentials['email']]);

        if (!$user) {
            // fail authentication with a custom error
            throw new CustomUserMessageAuthenticationException('Email could not be found.');
        }

        return $user;
    }

    public function checkCredentials($credentials, UserInterface $user)
    {
        return $this->passwordEncoder->isPasswordValid($user, $credentials['password']);
    }

    /**
     * Used to upgrade (rehash) the user's password automatically over time.
     */
    public function getPassword($credentials): ?string
    {
        return $credentials['password'];
    }

    public function onAuthenticationSuccess(Request $request, TokenInterface $token, $providerKey)
    {
        if ($targetPath = $this->getTargetPath($request->getSession(), $providerKey)) {
            return new RedirectResponse($targetPath);
        }

        return new RedirectResponse($this->urlGenerator->generate('index', [], RouterInterface::ABSOLUTE_URL));
    }

    protected function getLoginUrl()
    {
        return $this->urlGenerator->generate('login_user');
    }
}
```

My `routes.yaml` looks like so:

```
api_docs:
    path: /docs
    controller: App\Controller\IndexController::apiDocs

register:
    path: /register
    controller: App\Controller\SecurityController::register
index:
    path: /
    controller: App\Controller\IndexController::create

login_user:
    path: /login
    controller: App\Controller\SecurityController::login

logout:
    path: /logout
    controller: App\Controller\SecurityController::logout

user_change_password:
    path: /user/change/password
    controller: App\Controller\SecurityController::changePassword

user_list_pastes:
    path: /user/list
    controller: App\Controller\UserController::allPastes

user_delete_paste:
    path: /user/delete/paste/{slug}
    controller: App\Controller\UserController::deletePaste

user_editd_paste:
    path: /user/edit/paste/{slug}
    controller: App\Controller\UserController::editPaste
    requirements:
        slug: "[a-zA-Z]+"

list_paste:
    path: /{slug}
    controller: App\Controller\IndexController::show
    requirements:
        slug: "[a-zA-Z]+"

list_paste_raw:
    path: /raw/{slug}
    controller: App\Controller\IndexController::showRaw
    requirements:
        slug: "[a-zA-Z]+"

list_paste_download_raw:
    path: /dl/{slug}
    controller: App\Controller\IndexController::downloadRaw

api_get_paste:
    path: /api/v1/{slug}
    controller: App\Api\ApiController::getRaw
    methods: GET

api_login_check:
    path: /api/login_check
```

My `security.yaml` file is like so:

```
security:
    encoders:
        App\Entity\User:
            algorithm: auto
    providers:
        app_user_provider:
            entity:
                class: App\Entity\User
                property: email
    firewalls:
        login:
            pattern:  ^/api/login
            stateless: true
            anonymous: true
            json_login:
                check_path: /api/login_check
                success_handler: lexik_jwt_authentication.handler.authentication_success
                failure_handler: lexik_jwt_authentication.handler.authentication_failure
        api:
            pattern:   ^/api
            stateless: true
            guard:
                authenticators:
                    - lexik_jwt_authentication.jwt_token_authenticator
        dev:
            pattern: ^/(_(profiler|wdt)|css|images|js)/
            security: false
        main:
            anonymous: lazy
            guard:
                authenticators:
                    - App\Security\LoginFormAuthenticator
            logout:
                path: logout
                target: /
                invalidate_session: true
            remember_me:
                secret:   '%kernel.secret%'
                lifetime: 604800 # 1 week in seconds
                path:     /
                always_remember_me: true

    access_control:
         - { path: ^/api/login, roles: IS_AUTHENTICATED_ANONYMOUSLY }
         - { path: ^/api,       roles: IS_AUTHENTICATED_FULLY }
         - { path: ^/admin, roles: ROLE_ADMIN }
         - { path: ^/profile, roles: ROLE_USER }
         - { path: ^/login$, roles: IS_AUTHENTICATED_ANONYMOUSLY }
         - { path: ^/register, roles: IS_AUTHENTICATED_ANONYMOUSLY }
         - { path: ^/user, roles: ROLE_USER }
```

Whenever I try to access `/login` or `/register`(in a browser or doing a curl) I get a redirect to the `/` path. 
Without this package it works fine, I can access the login and register route without an issue.

When I try to access any route that starts with `/api/` I get a

```
{"code":401,"message":"JWT Token not found"}
```
Which is expected and absolutely fine. 

The firewalls are clearly separate with no overlapping patterns.

I can only assume that either there's some regex problem somewhere or this package silently overwrites some routes(although I can't find any indication of that happening other than the obvious redirect).

I understand that this is not a Q&A forum, however, I've asked on stackoverflow, googled the problem even asked other people, live, for help. No success on either.

I've either misconfigured something(which is likely) or this is indeed a bug.

**Comment by chalasr** [comment]
@KepplerPl Sorry, I missed time to be able to investigate on your support issue. Did you find the problem?

**Comment by KepplerPl** [comment]
@chalasr Hi, I did. Turns out it wasn't related to this package itself, for one reason or another Symfony was hanging sessions. I couldn't find the cause, I just did a full wipe of the sessions storage environment(in my case the database) and all was well. 

I assumed it was this package simply because it was the last installed and it fit, at the time, all the symptoms of my problem.

I marked it as closed since it doesn't apply to this package at all.
